gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -278,18 +278,10 @@ protected void startUp() {
   }
 
   /**
-   * Method to submit a {@link Dag} to the {@link DagManager} and delete adhoc flowSpecs from the FlowCatalog after
-   * persisting it in the other addDag method called. The DagManager's failure recovery method ensures the flow will be
-   * executed in the event of downtime.
-   * @param flowSpec
-   * @param dag
-   * @param persist
-   * @param setStatus
-   * @throws IOException
+   * Delete adhoc flowSpecs from the {@link FlowCatalog} after (separately) persisting via {@link DagManager#addDag(Dag, boolean, boolean)}.
+   * This DagManager's failure recovery mechanisms ensure the flow will be executed, even in the event of downtime.
    */
-  public synchronized void addDagAndRemoveAdhocFlowSpec(FlowSpec flowSpec, Dag<JobExecutionPlan> dag, boolean persist, boolean setStatus)
-      throws IOException {
-    addDag(dag, persist, setStatus);
+  public synchronized void removeFlowSpecIfAdhoc(FlowSpec flowSpec) throws IOException {
     // Only the active dagManager should delete the flowSpec
     if (isActive) {
       deleteSpecFromCatalogIfAdhoc(flowSpec);
```

Comment:
the delete looks good here.
but should it be at https://github.com/apache/gobblin/blob/master/gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java#L829 also @phet

Reply:
It shouldn't be in GSJS because all hosts in multi-active scheduler state needed access to the flowSpec in catalog see this PR desc: https://github.com/apache/gobblin/pull/3846. We were running into `no spec found ` in the `DagActionStoreChangeMonitor` error if an inactive host deleted the `flowSpec` before active host processed CDC stream message. For MA execution we want the spec in catalog until all hosts have received the CDC stream message and retrieved spec -> need to think about how to achieve this

Reply:
ok, we are not removing it from flow catalog there, just removing it from scheduledFlowSpecs